py/vttest/local_database.py

```diff
@@ -123,6 +154,153 @@ def load_schema(self):
+      field_names: The list of the field names in the table.
+      rows: A list of tuples with each tuple containing
+          the string representations of the fields.
+          The order of the representation must match the order of the field
+          names listed in 'field_names'.
+    """
+
+    field_names_string = ','.join(field_names)
+    values_string = ','.join(['(' + ','.join(row) +')' for row in rows])
+    # We use "INSERT IGNORE" to ignore duplicate key errors.
+    insert_query = ('INSERT IGNORE INTO %s (%s) VALUES %s' %
+                    (table_name, field_names_string, values_string))
+    self.mysql_execute([insert_query], db_name)
+
+  def generate_random_field(self, table_name, field_type, field_allows_nulls):
+    """Generates a random field string representation.
+
+    By 'string representation' we mean a string that is suitable to be a part
+    of an 'INSERT INTO' SQL statement.
+
+    Args:
+      table_name: The name of the table that will contain the generated field
+          value. Only used for a descriptive exception message in case of
+          an error.
+      field_type: The field_type as given by a "DESCRIBE <table>" SQL statement.
+      field_allows_nulls: Should be 'true' if this field allows NULLS.
+
+    Returns:
+      The random field.
+
+    Raises:
+      Exception: If 'field_type' is not supported.
+    """
+
+    value = None
+    if field_type.startswith('tinyint'):
+      value = self.random_integer(field_type, 1)
+    elif field_type.startswith('smallint'):
+      value = self.random_integer(field_type, 2)
+    elif field_type.startswith('mediumint'):
+      value = self.random_integer(field_type, 3)
+    elif field_type.startswith('int'):
+      value = self.random_integer(field_type, 4)
+    elif field_type.startswith('bigint'):
+      value = self.random_integer(field_type, 8)
+    elif field_type.startswith('float'):
+      value = self.random_float()
+    elif field_type.startswith('double'):
+      value = self.random_double()
+    else:
+      raise Exception('Populating random data in field type: %s is not yet '
+                      'supported. (table: %s)' % (field_type, table_name))
+    if field_allows_nulls and self.true_with_probability(self.null_probability):
+      return 'NULL'
+    return value
+
+  def true_with_probability(self, true_probability):
+    """Returns a pseudo-random boolean.
+
+    Args:
+      true_probability: The probability to use for returning 'true'.
+    Returns:
+      The value 'true' is with probability 'true_probability'.
+    """
+
+    return self.rng.uniform(0, 1) < true_probability
+
+  def random_integer(self, field_type, num_bytes):
+    num_bits = 8*num_bytes
+    if field_type.endswith('unsigned'):
+      return '%d' % (self.rng.randint(0, 2**num_bits-1))
+    return '%d' % (self.rng.randint(-2**(num_bits-1), 2**(num_bits-1)-1))
+
+  def random_float(self):
+    # Interpret a random 32-bit string as a float.
```

Comment:
I think there are illegal values in there, like NAN... Maybe needs to be double-checked.